default/AI/ResourcesAI.py

```diff
@@ -117,6 +124,87 @@ def print_resources_priority():
     warnings.clear()
 
 
+def weighted_sum_output(outputs):
+    """Return a weighted sum of planetary output
+    :param outputs: (industry, research)
+    :return: weighted sum of industry and research
+    """
+    return outputs[0] + RESEARCH_WEIGHTING * outputs[1]
+
+
+def assess_protection_focus(pid):
+    """Return True if planet should use Protection Focus"""
+    this_planet = planetMap[pid]
+    sys_status = foAI.foAIstate.systemStatus.get(this_planet.systemID, {})
+    threat_from_supply = (0.25 * foAI.foAIstate.empire_standard_enemy_rating *
+                          min(2, len(sys_status.get('enemies_nearly_supplied', []))))
+    print "Planet %s has regional+supply threat of %.1f" % ('P_%d<%s>'%(pid, this_planet.name), threat_from_supply)
```

Comment:
Hmm, yes, and clearly I was even using that to_str as a template here.  I expect I did this after running into the trouble that made me change PlanetUtils, but it doesn't look to me like that issue should really arise here.  Unless I remember and verify the issue that made me code it like this I'll switch it (or just take out the statement, I think I'm done debugging that aspect of it).